proposals/declarative-secrets.md

```diff
@@ -0,0 +1,58 @@
+# Declarative Secrets
+
+## Motiviation
+
+Various projects exist in the wild that require various secrets for them to run properly. Exmaples include:
+
+- https://github.com/lostintangent/codespaces-langchain
+- https://github.com/openai/openai-quickstart-python
+- https://github.com/openai/openai-quickstart-node
+
+Today these projects have to include specific instructions e.g. in their README telling users where to procure what they need to run and then how to set it up as a secret or add it as an environment variable. This currently acts as an impediment to adoption and promotion of codespaces for these projects.
+
+## Goal
+
+Simplify using codespaces for these kinds of projects by including it as a first-class part of the codespace creation flow.
+
+## Proposal
+
+Add an optional `secrets` to the `devContainer.base.schema.json`. This will be used to declare the secrets needed within the devcontainer.
+
+Property | Type | Description
+--- | --- | ---
+`secrets` | `object` | The keys of this object are the names of the secrets that are in use.
```

Comment:
Should we note here the shape of what a key in this map should look like?

Since implementations of this will likely source these key as environment variables, I think it would make sense to align with what a valid env variable key looks like (in unix-like systems): https://stackoverflow.com/questions/2821043/allowed-characters-in-linux-environment-variable-names#2821183